InvenTree/plugin/base/barcodes/mixins.py

```diff
@@ -37,3 +54,334 @@ def scan(self, barcode_data):
         """
 
         return None
+
+    def scan_receive_item(self, barcode_data, user, purchase_order=None, location=None):
+        """Scan a barcode to receive a purchase order item.
+
+        It's recommended to use the receive_purchase_order_item method to return from this function.
+
+        Returns:
+            None if the barcode_data could not be parsed.
+
+            A dict object containing:
+                - on success:
+                    a "success" message and the received "lineitem"
+                - on partial success (if there's missing information):
+                    an "action_required" message and the matched, but not yet received "lineitem"
+                - on failure:
+                    an "error" message
+        """
+
+        return None
+
+    @staticmethod
+    def parse_ecia_barcode2d(barcode_data: str | list[str]) -> dict[str, str]:
+        """Parse a standard ECIA 2D barcode, according to https://www.ecianow.org/assets/docs/ECIA_Specifications.pdf"""
+
+        if not isinstance(barcode_data, str):
+            data_split = barcode_data
+        elif not (data_split := BarcodeMixin.parse_isoiec_15434_barcode2d(barcode_data)):
+            return None
+
+        barcode_fields = {}
+        for entry in data_split:
+            for identifier, field_name in ECIA_DATA_IDENTIFIER_MAP.items():
+                if entry.startswith(identifier):
+                    barcode_fields[field_name] = entry[len(identifier):]
+                    break
+
+        return barcode_fields
+
+    @staticmethod
+    def parse_isoiec_15434_barcode2d(barcode_data: str) -> list[str]:
+        """Parse a ISO/IEC 15434 bardode, returning the split data section."""
+        HEADER = "[)>\x1E06\x1D"
+        TRAILER = "\x1E\x04"
+
+        # some old mouser barcodes start with this messed up header
+        OLD_MOUSER_HEADER = ">[)>06\x1D"
+        if barcode_data.startswith(OLD_MOUSER_HEADER):
+            barcode_data = barcode_data.replace(OLD_MOUSER_HEADER, HEADER, 1)
+
+        # most barcodes don't include the trailer, because "why would you stick to
+        # the standard, right?" so we only check for the header here
+        if not barcode_data.startswith(HEADER):
+            return
+
+        actual_data = barcode_data.split(HEADER, 1)[1].rsplit(TRAILER, 1)[0]
+
+        return actual_data.split("\x1D")
+
+    @staticmethod
+    def get_supplier_parts(sku: str, supplier: Company = None, mpn: str = None):
+        """Get a supplier part from SKU or by supplier and MPN."""
+        supplier_parts = SupplierPart.objects
```

Comment:
I would suggest you start with `supplier_parts = SupplierPart.objects.all()` (and then see comment below)